examples/privatelink/main.tf

```diff
@@ -0,0 +1,68 @@
+terraform {
+  required_providers {
+    clickhouse = {
+      version = "0.0.5"
+      source  = "ClickHouse/clickhouse"
+    }
+  }
+}
+
+variable "organization_id" {
+  type = string
+}
+
+variable "token_key" {
+  type = string
+}
+
+variable "token_secret" {
+  type = string
+}
+
+provider clickhouse {
+  organization_id = var.organization_id
+  token_key       = var.token_key
+  token_secret    = var.token_secret
+}
+
+# register PrivateLink endpoint ids here
+resource "clickhouse_private_endpoint_registration" "private_endpoints" {
+  private_endpoints = [
+    {
+      cloud_provider = "aws"
+      id             = "vpce-abcdef12345678910"
+      region         = "us-east-1"
+    }
+  ]
+}
+
+resource "clickhouse_service" "service" {
+  name                      = "My PrivateLink Service"
+  cloud_provider            = "aws"
+  region                    = "us-east-1"
+  tier                      = "production"
+  idle_scaling              = true
+  password_hash             = "n4bQgYhMfWWaL+qgxVrQFaO/TxsrC4Is0V1sFbDwCgg=" # base64 encoded sha256 hash of "test"
+
+  ip_access = [
+    {
+      source      = "0.0.0.0"
+      description = "Test IP"
+    }
+  ]
+
+  min_total_memory_gb  = 24
+  max_total_memory_gb  = 360
+  idle_timeout_minutes = 5
+
+  # use registered endpoint ids
+  private_endpoint_ids = clickhouse_private_endpoint_registration.private_endpoints.private_endpoint_ids
+}
+
+output "service_endpoints" {
```

Comment:
we should probably add TF code examples for PL config(commented code?):

something like:
```
resource "aws_vpc_endpoint" "this" {
  vpc_id            = var.vpc_id
  service_name      = clickhouse_service.service.endpoints <- it should use regional property, not instance's property to pass this info.
  vpc_endpoint_type = "Interface"
  security_group_ids = [
    Var.security_group_id1,var.security_group_id2, var.security_group_id3,
  ]
  subnet_ids          = [var.subnet_id1,var.subnet_id2,var.subnet_id3]
  private_dns_enabled = false
}
```

Reply:
I'd want to see this in the example.

This example shows a current issue with this implementation - there is a dependency loop where this `vpc_endpoint` requires the `clickhouse_service` to be created already (maybe that is what the comment about using regional property is about?) but the `clickhouse_service` also depends on the generated ID of this endpoint being created first.

Reply:
Added